src/psyclone/tests/psyir/nodes/array_mixin_test.py

```diff
@@ -335,47 +369,80 @@ def test_get_lbound_expression_unknown_size(extent):
     assert lbnd.children[0].symbol is symbol
 
 
+@pytest.mark.parametrize("extent", [ArrayType.Extent.DEFERRED,
+                                    ArrayType.Extent.ATTRIBUTE])
+def test_get_ubound_expression_unknown_size(extent):
+    '''
+    Test the get_ubound_expression when we have the definition of the
+    array type but its dimensions are unknown.
+
+    '''
+    symbol = DataSymbol("my_symbol", ArrayType(INTEGER_TYPE,
+                                               [extent, extent]))
+    aref = ArrayReference.create(symbol, [_ONE.copy(), _ONE.copy()])
+    ubnd = aref.get_ubound_expression(1)
+    assert isinstance(ubnd, BinaryOperation)
+    assert ubnd.operator == BinaryOperation.Operator.UBOUND
+    assert ubnd.children[0].symbol is symbol
+
+
 def test_aref_to_aos_lbound_expression():
     '''
     Test the get_lbound_expression() method for an ArrayReference to an array
     of structures.
 
     '''
     sgrid_type = StructureType.create(
-        [("id", INTEGER_TYPE, Symbol.Visibility.PUBLIC)])
+        [("ID", INTEGER_TYPE, Symbol.Visibility.PUBLIC)])
     sgrid_type_sym = DataTypeSymbol("subgrid_type", sgrid_type)
     sym = DataSymbol("subgrids", ArrayType(sgrid_type_sym, [(3, 10)]))
-    one = Literal("1", INTEGER_TYPE)
     lbound = BinaryOperation.create(BinaryOperation.Operator.LBOUND,
-                                    Reference(sym), one)
+                                    Reference(sym), _ONE.copy())
     ubound = BinaryOperation.create(BinaryOperation.Operator.UBOUND,
-                                    Reference(sym), one.copy())
+                                    Reference(sym), _ONE.copy())
     array = ArrayReference.create(sym, [Range.create(lbound, ubound)])
     lbnd = array.get_lbound_expression(0)
     assert lbnd.value == "3"
 
 
+def test_aref_to_aos_ubound_expression():
```

Comment:
If you factor-out the common functionality then you can probably get rid of this test :-)

Reply:
Tests are combined now.